src/com/oltpbenchmark/DBWorkload.java

```diff
@@ -939,32 +956,16 @@ private static Results runWorkload(List<BenchmarkModule> benchList,
     DecimalFormat df = new DecimalFormat();
     df.setMaximumFractionDigits(2);
 
-    LOG.info("Throughput: " + r + " reqs/sec");
-    LOG.info("Num New Order transactions : " + numNewOrderTransactions + ", time seconds: " + time);
-    LOG.info("TPM-C: " + df.format(tpmc));
-    LOG.info("Efficiency : " + df.format(efficiency) + "%");
-
-    boolean displayEnhancedLatencyMetrics =
-      xmlConfig.containsKey("displayEnhancedLatencyMetrics") &&
-      xmlConfig.getBoolean("displayEnhancedLatencyMetrics");
-    PrintLatencies(workers, displayEnhancedLatencyMetrics);
-
-    int numTxnTypes = workConfs.get(0).getNumTxnTypes();
-    int[] totalRetries = new int[numTxnTypes];
-    int[] totalFailures = new int[numTxnTypes];
-    for (Worker w : workers) {
-      for (int i = 0; i < numTxnTypes; ++i) {
-        totalRetries[i] += w.getTotalRetries()[i];
-        totalFailures[i] += w.getTotalFailures()[i];
-      }
-    }
-    LOG.info(String.format("Total retries: %s Total failures: %s",
-                           Arrays.toString(totalRetries), Arrays.toString(totalFailures)));
-    return r;
+    StringBuilder resultOut = new StringBuilder();
+    resultOut.append("\n");
+    resultOut.append("================RESULTS================\n");
+    resultOut.append(String.format("%18s | %18.2f\n", "TPM-C", tpmc));
+    resultOut.append(String.format("%18s | %17.2f%%\n", "Efficiency", efficiency));
+    resultOut.append(String.format("%18s | %18.2f\n", "Throughput (req/s)", r.getRequestsPerSecond()));
```

Comment:
do we need the StringBuilder?
Can't we have all of this in a single String.format()?

Reply:
I think this is more readable